### PR TITLE
Ports the Stamina HUD

### PR DESCRIPTION
**Pull request by Tsar-Salat**
## About The Pull Request

This ports the Stamina indicator HUD. It works very closely to the way the health HUD asset currently works, just changing state based on a flat value of whatever damage we are measuring, in this case, `maxHealth`.

Ports:
- https://github.com/tgstation/tgstation/pull/53974
- https://github.com/tgstation/tgstation/pull/64011
- https://github.com/tgstation/tgstation/pull/71623

## Why It's Good For The Game

The screwyhud doll shows limb disabling, sure, but it leaves no way to actually interpret how much stamina damage you are suffering. The only way to actually tell when you are going to be downed is many sec/antag hours practicing or knowing the code damage.

tl;dr stupid arbitrary hud make hard think

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/user-attachments/assets/13cef97b-8ce4-44d1-b5e9-aa228dd50f4a

</details>

## Changelog
:cl: rkz, Mothblocks, jacklecroy, kierany9, itseasytosee
add: Adds a stamina indicator hud
fix: fixes 'ghost' item overlays
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
This is good, but the utility of the hud icons on the side for combat things are significantly limited by vurtue of the fact that during combat it is difficult to look at the side of your screen.

**Comment by Tsar-Salat** [comment]
> This is good, but the utility of the hud icons on the side for combat things are significantly limited by vurtue of the fact that during combat it is difficult to look at the side of your screen.

It is no different than the screwy hud for limbs.

Ideal? No. But incremental utility is better than no utility.

**Comment by Tsar-Salat** [comment]
I hate you linter

**Comment by PowerfulBacon** [comment]
If you have hallucinations can stamina damage instead go to the standard crit HUD

**Comment by Tsar-Salat** [comment]
> If you have hallucinations can stamina damage instead go to the standard crit HUD

https://github.com/BeeStation/BeeStation-Hornet/pull/11250

Hallucinations are in a godawful state rn can this get looked at, I really don't want to dig through that mess of a "fluff" file

**Comment by Rukofamicom** [comment]
> Ideal? No. But incremental utility is better than no utility.

I second this. 

An ideal HUD displays relevant elements over or under the player without obstructing their view of other things. I don't think the latter part of that is possible in SS13 because no matter what it will be covering something, and that something will be nearby if the hud element is near you. 

I have a very specific test for you - Apply miner's salve to yourself and then stamcrit yourself. If the miner's salve side effect (One I intend to implement a lot further in the near future) is working properly, the stamina damage should NOT show up because miner's salve is supposed to obscure any damage from appearing on the HUD and give a false sense of security.

**Review by Rukofamicom — CHANGES_REQUESTED** [review]
Show me evidence of the hud failing to function with miner's salve active and I will approve this.

**Review by Rukofamicom — APPROVED** [review]

**Comment by Rukofamicom** [comment]
Turns out the current implementation does not hide stamina damage so this doesn't need to either. 

I'd still like if it were hidden by the effect that the salve applies, but nothing is being broken by it and I can just change that later myself.
